crates/aranya-client/tests/tests.rs

```diff
@@ -981,6 +981,50 @@ async fn test_setup_default_roles_rejects_unknown_owner() -> Result<()> {
     Ok(())
 }
 
+/// Tests that role creation works.
+#[test(tokio::test(flavor = "multi_thread"))]
+async fn test_create_role() -> Result<()> {
+    let mut devices = DevicesCtx::new("test_create_role").await?;
+
+    let team_id = devices.create_and_add_team().await?;
+    let owner_team = devices.owner.client.team(team_id);
+    let owner_role = owner_team
+        .roles()
+        .await?
+        .into_iter()
+        .find(|r| r.name == "owner")
+        .ok_or_else(|| anyhow::anyhow!("no owner role!?"))?;
+
+    owner_team
+        .create_role(text!("test_role"), owner_role.id)
+        .await?;
+
+    owner_team
+        .roles()
+        .await?
+        .into_iter()
+        .find(|r| r.name == "test_role")
+        .ok_or_else(|| anyhow::anyhow!("no test role found"))?;
+
+    Ok(())
+}
+
+/// Tests that role deletion works.
+#[test(tokio::test(flavor = "multi_thread"))]
+async fn test_delete_role() -> Result<()> {
+    let mut devices = DevicesCtx::new("test_delete_role").await?;
+
+    let team_id = devices.create_and_add_team().await?;
+    let roles = devices
+        .setup_default_roles_without_delegation(team_id)
+        .await?;
+
+    let owner_team = devices.owner.client.team(team_id);
+    owner_team.delete_role(roles.member().id).await?;
```

Comment:
The internal policy error may not be as clear as:
```suggestion
    owner_team.delete_role(roles.member().id).await.expect("expected to delete role");
```